Source/AppLovinAdapter.swift

```diff
@@ -48,25 +45,40 @@ final class AppLovinAdapter: PartnerAdapter {
             log(.setUpFailed(error))
             return completion(error)
         }
-        guard let sdk = ALSdk.shared(withKey: sdkKey) else {
+        let initConfig = ALSdkInitializationConfiguration(sdkKey: sdkKey) {builder in
+            builder.mediationProvider = "Chartboost"
+            if AppLovinAdapterConfiguration.testMode {
+                let idfa = ASIdentifierManager.shared().advertisingIdentifier
+                    if idfa.uuidString != "00000000-0000-0000-0000-000000000000" {
+                        builder.testDeviceAdvertisingIdentifiers = [idfa.uuidString]
+                    }
+                }
+                else {
+                        builder.testDeviceAdvertisingIdentifiers = []
+                }
+        }
+        
+        guard let sdk = ALSdk.shared() else {
             let error = error(.initializationFailureInvalidCredentials, description: "Invalid \(String.sdkKey)")
```

Comment:
at this point we haven't passed the credentials to the ALSdk so it doesn't seem like a returned nil would be caused by that.
Can we change the message by something like "ALSdk.shared() returned a nil value"?